docs/source/features/benchmark-and-generator.rst

```diff
@@ -0,0 +1,234 @@
+.. _benchmark:
+
+=================================
+Benchmark and Workload Generator
+=================================
+`AIBrix Benchmark <https://github.com/vllm-project/aibrix/tree/main/benchmarks>`_ contains the following components:
+
+- Dataset (Prompt) Generation
+- Workload Generation
+- Benchmark Client
+- Benchmark Scenarios (Work-in-progress)
+
+The diagram below shows the end-to-end steps of AIBrix benchmarks. Our components are highlighted as green rectangles in this diagram.
+  
+.. figure:: ../assets/images/benchmark/aibrix-benchmark-component-doc.png
+  :alt: benchmark-component-doc
+  :width: 100%
+  :align: center
+
+
+At its core, the AIBrix benchmark framework is built around a cleanly decoupled architecture: dataset generation, workload shaping, and benchmark execution. Each component can be customized independently, making it easy to plug in your own prompt logs, traffic traces, or experimental workloads—whether you're working on a new model deployment, scaling policy, or runtime optimization.
+
+Run AIBrix Benchmark End-to-End
+--------------------------------------
+
+.. note::
+    The benchmark script `benchmark.py <https://github.com/vllm-project/aibrix/blob/main/benchmarks/benchmark.py>`_ performs all steps up to the AIBrix workload format and trigger benchmark client without setting up benchmark environment for different scenarios. It assumes that AIBrix is already set up and expects a fully responsive endpoint.
+ 
+
+
+First, make sure you have configured your API key as well as your endpoint like
+
+.. code-block:: bash
+
+    export api_key="<your_api_key>"
+    kubectl -n envoy-gateway-system port-forward service/<gateway_service> 8888:80 &
+
+
+To run all steps using the default setting, try
+
+.. code-block:: bash
+
+    python benchmark.py all --config config.yaml
```

Comment:
do we still have `all` there? should it follow the argsparse pattern? like --stage all or --xxx